### DISCUSSION TITLE
Toast behavior inquiry

### DISCUSSION THREAD

**Discussion by arodriguezcortes** [twilio-labs/paste#2650] [General] [answered]
Hi Paste team. Until now I've been using your [recommendation](https://paste.twilio.design/components/toast/#use-an-alert-dialog-or-modal-when) on toasts:

> Allow toasts to persist on a page until a user closes them, especially if the toast has a link or button. Success toasts are an exception. Add an automatic time-out to a success toast if the only thing it does is confirm that the action was completed.

I set success toasts to last 6 seconds, and error toast to be closed by the user. However, my team has noticed that error toasts persist across pages (given I'm using a global toaster for the whole app), which I consider a good UX behavior. They suggested me to automatically close all toasts when the user navigates to another page, but that seems odd to me, because in my opinion, the user should acknowledge they was notified of the error.

So what should be the behavior for error toasts when the user leaves the page that triggered those toasts? should we persists them or clear them?

**Comment by SiTaggart** [comment, marked as answer]
Hi @arodriguezcortes 

That's a great question, and I think your reasoning is perfect.

If your error message is important, you should definitely allow people to see and read the error message in their own time. That might especially be significant in single page application, with client side routing. We often might perform a redirect upon or after an error has occurred, and you'd certainly not want to lose that error message just because we navigated you away to where you came from.

Obviously there is going to be some nuance in every decision, it probably can't be a blanket rule. Our general guidance is to keep them around until the user dismisses the error, but ultimately you know your user's and specific situation better than we do. Choose what's best for your customer and customer experience.

**Reply by arodriguezcortes** [reply]
Great, thanks @SiTaggart .
